docs/src/test-auth-js.md

```diff
@@ -278,6 +278,38 @@ test.describe(() => {
 });
 ```
 
+### Testing multiple roles together
+
+If you need to test how multiple authenticated roles interact together, use multiple [BrowserContext]s and [Page]s with different storage states in the same test. Any of the methods above to create multiple storage state files would work.
+
+```js tab=js-ts
+import { test } from '@playwright/test';
+
+test('two users chatting', async ({ browser }) => {
+  // page1 is signed in as user-1.
+  const page1 = await browser.newPage({ storageState: 'userStorageState-1.json' });
+
+  // page2 is signed in as user-2.
+  const page2 = await browser.newPage({ storageState: 'userStorageState-2.json' });
```

Comment:
Instead of using `browser.newPage` which has implicit `context` creation, let's make the example help the user understand `context`s and be a bit more verbose showing the `context`s created explicitly:


```ts
const adminUserContext = await browser.newContext({ ...contextOptions, storageState: 'userStorageState-admin.json' })
const adminUserPage = await browser.newContext({ ...contextOptions, storageState: 'userStorageState-adminUser.json' });

const standardUserContext = await browser.newContext({ ...contextOptions, storageState: 'userStorageState-standardUser.json' })
const standardUserPage = await browser.newPage();
```